mods/weather/init.lua

```diff
@@ -0,0 +1,105 @@
+-- Disable by mapgen or setting
+
+local mg_name = minetest.get_mapgen_setting("mg_name")
+if mg_name == "v6" or mg_name == "singlenode" or
+		minetest.settings:get_bool("enable_weather") == false then
```

Comment:
```suggestion
		minetest.settings:get_bool("enable_weather") ~= true then
```

Reply:
This was intentional so that weather will be enabled if the setting is not defined (nil).